### DISCUSSION TITLE
How to use a model with external api? Not with database.

### DISCUSSION THREAD

**Discussion by awesome1128** [filamentphp/filament#15122] [Help]
### Package

Panel builder
Table builder
Form builder

### Package Version

v3.0.0

### How can we help you?

I am building an app now.
But I have super critical problem. 
For building an UI with table grid, I created a resource called SimcardManagement, but I realized that resources can be used with database only.
But all my data is in baserow.io, so I don't want to use Laravel mysql database. Meaning, I want to put a table grid and show data with calling baserow api.
For now, I created BaserowApi.php file in app/Models, but I am not sure how to use this in SimcardManagementResource.php file because as I mentioned above Resource works with Model related to the table of database.
This is my headache.
Please help me if someone know this.
Ask me if you guys don't understand what I want.
Thank you in advance.

- SimcardManagement Resource
![image](https://github.com/user-attachments/assets/064b4341-efd2-417a-bb60-35332c61d4ce)

- BaserowApi.php
![image](https://github.com/user-attachments/assets/5f79ac45-bbde-4941-82ff-8a3d2c196db9)

**Comment by leandrocfe** [comment]
You can work with [sushi](https://filamentphp.com/content/leandrocfe-how-to-consume-an-external-api-with-filament-tables)
You can [customize the creation/edition process](https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-the-creation-process)

**Reply by awesome1128** [reply]
Wow!
Looks great!
Let me check, thank you so much!

**Reply by awesome1128** [reply]
This is deprecated, right?
![image](https://github.com/user-attachments/assets/c5801f65-500f-4eaf-9258-bbbac08d356a)

**Reply by awesome1128** [reply]
This is deprecated, right?
Do you have updated document?
![image](https://github.com/user-attachments/assets/a7d0cb23-d3fe-4604-990c-8f7228f696c7)

**Reply by awesome1128** [reply]
I followed your document, so I got this error.
![image](https://github.com/user-attachments/assets/e1bfdc88-a6fd-42db-ad7f-dc2261294074)

**Reply by awesome1128** [reply]
In your introduction, there is no any logic to insert data to database, but as you can see, this is an Insert query error.
I don't understand why.

**Reply by awesome1128** [reply]
Ah, okay, I understand why.

**Reply by awesome1128** [reply]
I followed you on github, thank you for your help. 👍👍

**Reply by leandrocfe** [reply]
Yes, because this article uses Filament v2, but you can follow the same idea

**Reply by awesome1128** [reply]
Hello, sorry.
I have another problem.

For example, I have a link in grid table, so if I click it, opens a new page with route parameters.
New page is opening correctly. (eg: http://localhost:869/admin/activate/2/19.99/3.28)
So, for a new page, I created a livewire component called SimcardActivate.
But I keep getting the following error message.
"Livewire page component layout view not found: [components.layouts.app]"

Actually, I want to open a modal with passing the parameters when I click a link on grid table.
Not Livewire component.

But for example, if I want to use livewire component, I want to put the component on A section with opening a new tab. See below image.
![image](https://github.com/user-attachments/assets/32942dcd-d9c3-4e60-aa74-7c87f85f572d)

Please guide me if possible, thank you in advance.

The screenshots for problems:
![image](https://github.com/user-attachments/assets/fd44bcbc-a96f-4c6a-aed5-d59f4408a7e8)
![image](https://github.com/user-attachments/assets/4dc243cf-53fc-4ae1-82ee-6bb1b0cc6090)
![image](https://github.com/user-attachments/assets/14383b5f-c4d4-4618-bc4d-65c1df0dfce7)
![image](https://github.com/user-attachments/assets/0f90cc99-8763-4e10-a42b-42f895d0a74b)
